### ISSUE TITLE
Allow reloading of config and restart validation in server mode on Unix.

### ISSUE THREAD

**Issue by partim** [NLnetLabs/routinator#200]
Reloading the config should happen via SIGHUP, restarting validation should happen through SIGUSR1.

**Comment by hellerve** [comment]
Do you have a preferred library in mind here?

**Comment by partim** [comment]
Not really.

Config reload will require some rewiring. Probably best to just shut everything down, reload the config, and start up again.

**Comment by hellerve** [comment]
Yeah I’ve been playing around with a bunch of libraries (most notably rust-simple-signal and Tokio’s mostly undocumented Unix signal handling), and there is some lifetime trickery involved that makes it a little harder than just calling `repo.reload_tals()` (as an example) in a handler function!

**Comment by partim** [comment]
I think the best way is to use the pipe approach and ‘select’ for that instead of the current `idle.wait()` in `Server::run`.
